tests/_utils-tests/model.js

```diff
@@ -256,32 +261,33 @@ describe( 'model test utils', () => {
 				selection.collapse( root );
 
 				expect( stringify( root, selection ) ).to.equal(
-					'<selection /><a></a>foo<$text bold=true>bar</$text><b></b>'
+					'[]<a></a>foo<$text bold="true">bar</$text><b></b>'
 				);
 			} );
 
 			it( 'writes selection collapsed in a text', () => {
 				selection.collapse( root, 3 );
 
 				expect( stringify( root, selection ) ).to.equal(
-					'<a></a>fo<selection />o<$text bold=true>bar</$text><b></b>'
+					'<a></a>fo[]o<$text bold="true">bar</$text><b></b>'
 				);
 			} );
 
 			it( 'writes selection collapsed at the text left boundary', () => {
 				selection.collapse( elA, 'after' );
 
 				expect( stringify( root, selection ) ).to.equal(
-					'<a></a><selection />foo<$text bold=true>bar</$text><b></b>'
+					'<a></a>[]foo<$text bold="true">bar</$text><b></b>'
 				);
 			} );
 
 			it( 'writes selection collapsed at the text right boundary', () => {
 				selection.collapse( elB, 'before' );
 
 				expect( stringify( root, selection ) ).to.equal(
-					'<a></a>foo<$text bold=true>bar</$text><selection bold=true /><b></b>'
+					'<a></a>foo<$text bold="true">bar</$text>[]<b></b>'
 				);
+				expect( selection.getAttribute( 'bold' ) ).to.true;
```

Comment:
You do not check if there is no other selection attributes. Add check to the number of attributes.